apps/backend/src/adoption/adoption.interface.ts

```diff
@@ -0,0 +1,12 @@
+import { AdoptionType } from '@animavita/models';
+import { Location } from '../user/repositories/mongodb/user.interface';
+
+export type AdoptionDto = AdoptionType;
+
+export type AdoptionResponse = AdoptionType & {
+  user: {
+    id: string;
```

Comment:
After skimming mongodb documentation, I found this ObjectId type, does it make sense to use it here?
https://www.mongodb.com/docs/drivers/node/current/fundamentals/typescript/#insert-operations-and-the-_id-field

Reply:
We have a few types of interfaces; in this case, it's just the interface used in the service response. So basically it's not related to Mongo itself but to the JSON schema to be returned. Now if you look at [adoption-schema](https://github.com/animavita/animavita/blob/v2/apps/backend/src/adoption/repositories/mongodb/adoption-mongo.schema.ts) we do have a reference to the user using object id type.